src/item_factory.cpp

```diff
@@ -2670,6 +2736,55 @@ static int parse_acc_string( const std::string &to_hit )
     return acc;
 }
 
+int parse_acc_string( const std::string &to_hit )
+{
+    std::string grip, length, surface, balance;
+    size_t last_split = 0;
+    size_t split_len = to_hit.find( "_", last_split );
+
+    if( split_len != std::string::npos && split_len > 0 ) {
+        grip = to_hit.substr( last_split, split_len );
+    }
+    last_split = split_len + 1;
+    split_len = to_hit.find( "_", last_split );
+    if( split_len != std::string::npos && split_len > 0 ) {
+        length = to_hit.substr( last_split, split_len - last_split );
+    }
+    last_split = split_len + 1;
+    split_len = to_hit.find( "_", last_split );
+    if( split_len != std::string::npos && split_len > 0 ) {
+        surface = to_hit.substr( last_split, split_len - last_split );
+    }
+    last_split = split_len + 1;
+    split_len = to_hit.size();
+    if( last_split < split_len ) {
+        balance = to_hit.substr( last_split, split_len );
+    }
+    return calc_acc_values( grip, length, surface, balance );
+}
+
+struct acc_data {
+    grip_val grip = 0;
+    length_val length = 0;
+    surface_val surface = 0;
+    balance_val balance = 0;
+
+    int sum_values() {
+        return static_cast<int>( grip ) + static_cast<int>( length ) +
+	       static_cast<int>( surface ) + static_cast<int>( balance ) - 2;
+    }
+    void load( JsonObject &jo );
```

Comment:
You need a `deserialize( JsonIn &jo )` function which forwards to this for this to work with mandatory/optional.

Reply:
Thanks.